components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt

```diff
@@ -471,9 +491,16 @@ class QrFragment : Fragment() {
 
         private const val CAMERA_CLOSE_LOCK_TIMEOUT_MS = 2500L
 
-        fun newInstance(listener: OnScanCompleteListener): QrFragment {
+        /**
+         * Returns a new instance of QR Fragment
+         * @param listener Listener invoked when the QR scan completed successfully.
+         * @param scanMessage (Optional) Scan message to be displayed.
+         */
+        fun newInstance(listener: OnScanCompleteListener, scanMessage: Int? = null): QrFragment {
             return QrFragment().apply {
+                retainInstance = true
```

Comment:
Was this set to keep the new `scanMessage` when the fragment gets re-created? If so, we already solve that for the `scanCompleteListener` i.e. the feature could just set the `scanMessage` onStart. Then we don't have to retain the instance: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFeature.kt#L52

Reply:
I had some problems with testing in Fenix. It would either not show the message in the first draw, or it would not change it on rotation or not changing the StaticLayout width on rotation. Maybe it has to do with Fenix declaring the handling of config changes?

Reply:
I am not sure, but passing along `scanMessage` seems preferable over keeping the instance of the fragment around. Let's try / debug the UI issue you're seeing if still present?

Reply:
I remembered the other way around. Removing `retainInstance = true` has no implications in Fenix, but in sample app the message disappears on rotation. I think Fenix retains instance by default, but other clients might not.

Reply:
OK, but that's an even stronger reason to remove it ;). If it's fine in Fenix, we can debug the UI hiccup in sample browser later. So, I would treat `scanMessage` the same as `scanCompleteListener` and make sure we set it when the features is started. I don't think we should keep the fragment instance longer than we need to.

Reply:
I wasn't thinking about sample, but other clients that would use the QR scan feature.

Reply:
I've removed it. We can pick it up at a latter time if the message disappearing issue arises.

Reply:
@mcarare but we still need to pass the `scanMessage` in `onStart` similar to: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFeature.kt#L52

Otherwise, you will lose the string if the fragment gets re-created which might be exactly the problem you run into in sample browser?

Reply:
Yes, but in Fenix the fragment does not get recreated.

Reply:
The message still disappears even with `   it.scanMessage = scanMessage` added in `start()`. Only calling setupMessage again could fix that, but it would have to be synchronized somehow with the viewfinder redrawing.

Reply:
Ok, can you add the change here? I think we want that either way. I can take a quick look later today and worst case we can investigate as a follow-up if we can't get it to work, but let's definitely make sure we have the message in the fragment when it re-starts.

Reply:
Sorry, do you mean add `retainInstance = true` here? `this.scanMessage = scanMessage` is already called in `newInstance()`.

Reply:
@mcarare ah no, I mean passing along the `scanMessage` in onStart: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFeature.kt#L52

Reply:
OK, yeah, making the message static as in your latest change works too! Tested with "don't keep activities" and the custom finder and message are still fine after the activity gets recreated.